src/leappto/cli.py

```diff
@@ -135,27 +142,64 @@ def _set_ssh_config(username, identity, use_sshpass=False):
         return use_sshpass, ssh_options
 
     class MigrationContext:
-        def __init__(self, target, ssh_settings, disk, forwarded_ports=None):
+
+        _SSH_CTL_PATH = '{}/.ssh/ctl'.format(os.environ['HOME'])
+        _SSH_CONTROL_PATH = '-o ControlPath="{}/%L-%r@%h:%p"'.format(_SSH_CTL_PATH)
+
+        def __init__(self, source, target, ssh_settings, disk, forwarded_ports=None, rsync_cp_backend=False):
+            self.source = source
             self.target = target
-            self.use_sshpass, self.target_cfg = ssh_settings
+            self.use_sshpass, self.ssh_cfg = ssh_settings
```

Comment:
I'm a bit on the fence by this rename - I know that the current implementation of `MigrationContext` is not exactly right, but the thing is that we're moving away from the Libvirt/Vagrant workflow towards arbitrary source/target workflow and in that sense I think it would be best to introduce a pair of variables (`self.target_cfg` and `self.source_cfg`) - each of them facilitating access to the respective machine, since each machine can have different SSH config/keys, and this change would be also well aligned with the below addition of `{source,target}_addr` properties.